apps/explorer/src/api/assets/assetsApi.ts

```diff
@@ -157,7 +164,27 @@ export const assetsQueries = {
       queryOptions({
         queryKey: ['nftData', dataUri],
         queryFn: (): Promise<NFTTokenUriMetaData & { assetId: string }> | undefined =>
-          fetch(dataUri).then((res) => res.json().then((f) => ({ ...f, assetId }))),
+          fetch(dataUri).then(async (res) => {
+            let data: NFTTokenUriMetaData | undefined
+            try {
+              data = (await res.json()) as NFTTokenUriMetaData
+
+              if (!matchesNFTTokenUriMetaDataSchema(data)) {
+                data = undefined
+                return Promise.reject()
+              }
+            } catch (e) {
+              if (dataUri.startsWith('data:application/json;utf8,')) {
+                data = JSON.parse(dataUri.split('data:application/json;utf8,')[1])
+              }
```

Comment:
This also needs to be checked if it matches the schema

Reply:
Moreover, I believe that by using `axios`, we don't need to do this weird `startsWith('data` thing. See how it's done in the desktop wallet tanstack branch: https://github.com/alephium/alephium-frontend/blob/next-tanstack/apps/desktop-wallet/src/api/queries/tokenQueries.ts#L101-L119

Reply:
Done with Axios 👍